level_elements/level_logic.gd

```diff
@@ -166,11 +176,24 @@ func set_glitch_color(new_glitch_color: Enums.colors, is_undo := false) -> void:
 	# so that it doesn't have to go through ALL all? 
 	
 	for key: Key in level.keys.get_children():
-		# TODO/PERF: no!
+		# better?
+		if key.key_data.color != Enums.colors.glitch:
+			continue
 		key._on_changed_glitch_color()
 	
 	for door: Door in level.doors.get_children():
 		var _door_data := door.door_data
+		var has_glitch := false
+		if _door_data.outer_color == Enums.colors.glitch:
+			has_glitch = true
+		else:
+			for lock in _door_data.locks:
+				if lock.color == Enums.colors.glitch:
+					has_glitch = true
+					break
+		if not has_glitch:
+			# better?
```

Comment:
The issue was with going through all doors. but with this we're going through all doors _and_ their locks? D: you do save on useless undo commands, but, ideally you'd have neither. I think I'll keep the simpler code until I get around to the proper solution